src/name/bizna/ocarmsim/FakeMachine.java

```diff
@@ -160,12 +170,14 @@ public boolean stop() {
 
 	@Override
 	public boolean signal(String name, Object... args) {
-		synchronized(signals) {
-			signals.add(new SimSignal(name, args));
+		if (OCARM.instance.shouldTraceInvocations()) {
+			OCARM.logger.info("signal pushed: %s", name);
 		}
-		synchronized(thread) {
-			thread.notify();
```

Comment:
So the new simulator always has a `Debugger` instance, and that `Debugger` must call `notify()`?